consensus/replay.go

```diff
@@ -290,18 +314,18 @@ func (h *Handshaker) ReplayBlocks(appHash []byte, appBlockHeight int, proxyApp p
 	return nil, nil
 }
 
-func (h *Handshaker) replayBlocks(proxyApp proxy.AppConns, appBlockHeight, storeBlockHeight int, useReplayFunc bool) ([]byte, error) {
+func (h *Handshaker) replayBlocks(proxyApp proxy.AppConns, appBlockHeight, storeBlockHeight int, mutateState bool) ([]byte, error) {
 	// App is further behind than it should be, so we need to replay blocks.
 	// We replay all blocks from appBlockHeight+1.
-	// If useReplayFunc == true, stop short of the last block
-	// so it can be replayed using the WAL in ReplayBlocks.
 	// Note that we don't have an old version of the state,
-	// so we by-pass state validation using sm.ApplyBlock.
+	// so we by-pass state validation/mutation using sm.ApplyBlock.
+	// If mutateState == true, stop short of the last block
+	// so it can be replayed with a real state.ApplyBlock
```

Comment:
This comment isn't perfectly correct... it doesn't just stop short of the last block, but does the replaying itself.